### PR TITLE
feat: Add fetchTaskRequestById

### LINKED ISSUE #1273: Fetch route for task request using query params
## Description
Currently task requests api only has fetch request that fetches all the data at once, add a way for this api to accept query params to call individual task request by their respective id.

## Proposed Solution
Add route containing a slug which will be taskRequestId and fetch the individual taskRequest from the collection.

### PR DESCRIPTION
**Pull request by TanishqSingla**
closes #1273 

## Note to reviwer
The following PR contains the tests for the model which have been separated from this PR.
#1300 

## Screenshots
&nbsp;<details><summary>Show Screenshot</summary>Screenshot from 2023-07-18 20-42-08</details>

### CODE REVIEW

**Comment by TanishqSingla** [comment]
Files changed and their respective coverage:
|File                                   | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
|---------------------------------------|---------|----------|---------|---------|-------------------|
|   taskRequests.js                      |     100 |      100 |     100 |     100 |    
|   tasksRequests.js                     |   84.78 |    92.85 |     100 |   84.78 |

File: models/taskRequests.js [hunk 1]
```diff
@@ -39,6 +41,23 @@ const fetchTaskRequests = async () => {
...
+  const taskRequestSnapshot = await taskRequestsCollection.doc(taskRequestId).get();
+  const taskRequestData = taskRequestSnapshot.data();
+
+  if (taskRequestData) {
+    taskRequestData.id = taskRequestSnapshot.id;
+    taskRequestData.url = new URL(`/taskRequests/${taskRequestData.id}`, config.get("services.rdsUi.baseUrl"));
+  }
+  return taskRequestData;
+};
+
```
Comment (fakhruddinkw) [inline models/taskRequests.js:60]: Why is there no try catch?

**Review by fakhruddinkw — COMMENTED** [review]
please also add unit tests.

File: models/taskRequests.js (hunk 1 above)
Comment (TanishqSingla) [reply models/taskRequests.js:60]: Currently there is no logger for our backend and we're just throwing the error in the catch statement, so having a catch here will be redundant as we're already catching it in the controller.

**Review by fakhruddinkw — APPROVED** [review]

File: models/taskRequests.js
[the remark this answers is not in the document]
Comment (TanishqSingla) [reply models/taskRequests.js:1]: mentioned

**Review by vinit717 — DISMISSED** [review]

**Review by fakhruddinkw — APPROVED** [review]

File: controllers/tasksRequests.js [hunk 2]
```diff
@@ -23,6 +23,26 @@ const fetchTaskRequests = async (_, res) => {
   }
 };
 
+const fetchTaskRequestById = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const data = await taskRequestsModel.fetchTaskRequestById(id);
+
+    if (!data.taskRequestExists) {
```
Comment (iamitprakash) [inline controllers/tasksRequests.js:31]: here data object returned from the taskRequestsModel.fetchTaskRequestById() method contains a property called taskRequestExists. This property is true if the task request was found, and false if it was not found. However, the code you provided does not check the value of this property before returning the response. This means that if the task request was not found, the code will still return a status code of 200

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: controllers/tasksRequests.js (hunk 2 above)
Comment (TanishqSingla) [reply controllers/tasksRequests.js:31]: I am specifically testing for `taskRequestExists` key's value, and I have also written a test for the same, which you can find in #1300

Comment (iamitprakash) [reply controllers/tasksRequests.js:31]: @RitikJaiswal75 can you look here please

Comment (RitikJaiswal75) [reply controllers/tasksRequests.js:31]: I cant find this test in #1300 (looking for it in integration test)

Comment (TanishqSingla) [reply controllers/tasksRequests.js:31]: @RitikJaiswal75 Can you specify what exactly are you looking for

File: models/taskRequests.js [hunk 3]
```diff
@@ -13,32 +13,66 @@ const userModel = require("./users");
...
+ * @return Promise<{taskRequest: Object}>
+ */
+const fetchTaskRequestById = async (taskRequestId) => {
+  try {
+    const taskRequestSnapshot = await taskRequestsCollection.doc(taskRequestId).get();
+    const taskRequestData = taskRequestSnapshot.data();
+
+    if (taskRequestData) {
+      taskRequestData.id = taskRequestSnapshot.id;
+      taskRequestData.url = new URL(`/taskRequests/${taskRequestData.id}`, config.get("services.rdsUi.baseUrl"));
```
[the remark this answers is not in the document]
Comment (TanishqSingla) [reply models/taskRequests.js:61]: I am actually sending resource `id` along with `url`, so line 60 creates the `id` key in object.
You can refer this doc: https://cloud.google.com/static/files/apigee/apigee-web-api-design-the-missing-link-ebook.pdf

File: controllers/tasksRequests.js [hunk 4]
```diff
@@ -6,8 +6,8 @@ const fetchTaskRequests = async (_, res) => {
   try {
     const data = await taskRequestsModel.fetchTaskRequests();
 
-    if (data || data.length > 0) {
-      res.status(200).json({
+    if (data.length > 0) {
```
Comment (iamitprakash) [inline controllers/tasksRequests.js:9]: what if the data is null here, data.length won't give you error...

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**